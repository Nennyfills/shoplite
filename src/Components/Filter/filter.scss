.filter {
  width: 20em;
  select {
    width: 100%;
    font-size: 1rem;
    -webkit-appearance: none;
    appearance: none;
    outline: none;
    border-radius: 2px;
    border: 1px solid #ccc;
    background: url(http://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/br_down.png)
      no-repeat 92%;
    padding: 1em 1em;
  }
}
