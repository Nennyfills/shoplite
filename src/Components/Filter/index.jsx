import React from 'react';
import PropTypes from 'prop-types';

import './filter.scss';

const Filter = ({ products, currencies, onChange }) => {
  return (
    <div className="filter" data-test="filter">
      <select
        data-test="filter-selector"
        type="text"
        id="filter-input"
        list="products"
        name="option"
        onChange={onChange}
      >
        <option>{currencies ?? 'Filter by'}</option>
        {products?.map((item) => (
          <option
            value={currencies ? item : item.title}
            key={currencies ? item : item.title}
          >
            {currencies ? item : item.title}
          </option>
        ))}
      </select>
    </div>
  );
};

Filter.defaultProps = {
  loading: true,
  products: [],
  currencies: null,
};

Filter.propTypes = {
  onChange: PropTypes.func.isRequired,
  products: PropTypes.array,
  currencies: PropTypes.string,
};
export default Filter;
