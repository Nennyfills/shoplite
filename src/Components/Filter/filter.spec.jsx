import React from 'react';
import { shallow } from 'enzyme';

import { checkProps } from 'Test/propsTypes';
import Filter from '.';

const props = {
  onChange: jest.fn(),
  products: ['NGN', 'EROS'],
  currencies: 'NGN',
};

const setUp = (props = {}) => {
  const component = shallow(<Filter {...props} />);
  return component;
};

describe('Filter', () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });

  it('Should render without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should render filter div', () => {
    const wrapper = component.find(`[data-test='filter']`);
    expect(wrapper.length).toBe(1);
  });

  it('Should render drop down button', () => {
    const wrapper = component.find(`[data-test='filter-selector']`);
    expect(wrapper.length).toBe(1);
  });

  it('Should simulate currency change', () => {
    const e = {
      target: {
        value: 'NGN',
      },
    };
    const select = component.find(`[data-test='filter-selector']`);
    select.simulate('change', e);
    expect(props.onChange.mock.calls.length).toBe(1);
  });

  it('Checking PropTypes should not throw error', () => {
    const propErr = checkProps(Filter, props);
    expect(propErr).toBeUndefined();
  });
});
