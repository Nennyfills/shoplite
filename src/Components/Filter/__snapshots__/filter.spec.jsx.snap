// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Should render without errors 1`] = `ShallowWrapper {}`;
